lib/CXGN/BrAPI/v2/ObservationVariables.pm

```diff
@@ -308,45 +309,53 @@ sub store {
-            bcs_schema          => $self->bcs_schema,
-            external_references => $params->{externalReferences} || [],
-            table_name          => "cvterm",
-            table_id_key        => "cvterm_id",
-            id                  => $cvterm_id
-        });
-        my $trait = CXGN::Trait->new({ bcs_schema => $self->bcs_schema,
-            cvterm_id                             => $cvterm_id,
-            name                                  => $name,
-            ontology_id                           => $ontology_id,
-            definition                            => $description,
-            synonyms                              => $synonyms,
-            external_references                   => $external_references,
-            method                                => $method,
-            scale                                 => $scale
-        });
-        $trait->{active} = $active;
+    my $coderef = sub {
+        foreach my $params (@{$data}) {
+            my $cvterm_id = $params->{observationVariableDbId} || undef;
+            my $name = $params->{observationVariableName};
+            my $ontology_id = $params->{ontologyReference}{ontologyDbId};
+            my $description = $params->{trait}{traitDescription};
+            my $synonyms = $params->{synonyms};
+            my $active = $params->{status} ne "archived";
+
+            #TODO: Parse this when it initially comes into the brapi controller
+            my $scale = CXGN::BrAPI::v2::Scales->new({
+                bcs_schema => $self->bcs_schema,
+                scale      => $params->{scale}
+            });
+            my $method = CXGN::BrAPI::v2::Methods->new({
+                bcs_schema => $self->bcs_schema,
+                method     => $params->{method}
+            });
+            my $external_references = CXGN::BrAPI::v2::ExternalReferences->new({
+                bcs_schema          => $self->bcs_schema,
+                external_references => $params->{externalReferences} || [],
+                table_name          => "cvterm",
+                table_id_key        => "cvterm_id",
+                id                  => $cvterm_id
+            });
+            my $trait = CXGN::Trait->new({ bcs_schema => $self->bcs_schema,
+                cvterm_id                             => $cvterm_id,
+                name                                  => $name,
+                ontology_id                           => $ontology_id,
+                definition                            => $description,
+                synonyms                              => $synonyms,
+                external_references                   => $external_references,
+                method                                => $method,
+                scale                                 => $scale
+            });
+            $trait->{active} = $active;
+
+            my $variable = $trait->store();
```

Comment:
I see there's already a transaction for $trait->store(). I wonder what was missing from that compared to here for it to not work?

Reply:
It would roll back the transaction for that single trait, but if you send 10 traits and the 6th one fails, the previous 5 still save.

Reply:
ah yes I see now. Thanks